### ISSUE TITLE
Studio: Improving graph rendering to split vertices and edges

### ISSUE THREAD

**Issue by lvca** [orientechnologies/orientdb#6214]
Collect only the edges that are relevant to the vertices in the resultset.

Example:
```json
{
	"graph": {
		"vertices": [{
			"@rid": "#17:62",
			"@class": "Whisky",
			"name": "Springbank"
		}, {
			"@rid": "#17:67",
			"@class": "Whisky",
			"name": "Tobermory"
		}, {
			"@rid": "#15:3",
			"@class": "FlavourGroup",
			"dryness": "Medium",
			"name": "D",
			"description": "Low or No Peat, with Fruity, Floral, Malty Notes and Nutty Hints.",
			"body": "Light"
		}, {
			"@rid": "#15:5",
			"@class": "FlavourGroup",
			"dryness": "Medium",
			"name": "F",
			"description": "Medium-Bodied, Medium-Sweet, Low Peat, Malty Notes and Sherry, Honey, Spicy Hints.",
			"body": "Medium"
		}, {
			"@rid": "#12:1",
			"@class": "Location",
			"name": "Alexandria"
		}, {
			"@rid": "#17:2",
			"@class": "Whisky",
			"name": "Arran"
		}, {
			"@rid": "#12:6",
			"@class": "Location",
			"name": "Blackford"
		}, {
			"@rid": "#17:7",
			"@class": "Whisky",
			"name": "BenRiach"
		}, {
			"@rid": "#12:11",
			"@class": "Location",
			"name": "Dalmuir"
		}, {
			"@rid": "#17:12",
			"@class": "Whisky",
			"name": "Bruichladdich"
		}, {
			"@rid": "#12:16",
			"@class": "Location",
			"name": "Elgin"
		}, {
			"@rid": "#17:17",
			"@class": "Whisky",
			"name": "Cragganmore"
		}, {
			"@rid": "#12:21",
			"@class": "Location",
			"name": "Isle of Islay"
		}, {
			"@rid": "#14:3",
			"@class": "Flavour",
			"name": "Malt"
		}, {
			"@rid": "#17:22",
			"@class": "Whisky",
			"name": "Deanston"
		}, {
			"@rid": "#12:26",
			"@class": "Location",
			"name": "Kirkwall"
		}, {
			"@rid": "#14:8",
			"@class": "Flavour",
			"name": "Peat"
		}, {
			"@rid": "#17:27",
			"@class": "Whisky",
			"name": "Glen Grant"
		}, {
			"@rid": "#12:31",
			"@class": "Location",
			"name": "Muir of Ord"
		}, {
			"@rid": "#17:32",
			"@class": "Whisky",
			"name": "Glendronach"
		}, {
			"@rid": "#12:36",
			"@class": "Location",
			"name": "Oldmeldrum"
		}, {
			"@rid": "#11:4",
			"@class": "Region",
			"name": "Lowlands"
		}, {
			"@rid": "#17:37",
			"@class": "Whisky",
			"name": "Glenkinchie"
		}, {
			"@rid": "#12:41",
			"@class": "Location",
			"name": "Rothes"
		}, {
			"@rid": "#17:42",
			"@class": "Whisky",
			"name": "Glenturret"
		}, {
			"@rid": "#17:47",
			"@class": "Whisky",
			"name": "Knockando"
		}, {
			"@rid": "#17:52",
			"@class": "Whisky",
			"name": "Macallan"
		}, {
			"@rid": "#17:57",
			"@class": "Whisky",
			"name": "Old Pulteney"
		}],
		"edges": [{
			"@rid": "#13:11",
			"@class": "InRegion",
			"from": "#11:4",
			"to": "#12:11"
		}, {
			"@rid": "#16:30",
			"@class": "FlavourInGroup",
			"from": "#14:2",
			"to": "#15:5"
		}, {
			"@rid": "#13:16",
			"@class": "InRegion",
			"from": "#11:5",
			"to": "#12:16"
		}, {
			"@rid": "#16:35",
			"@class": "FlavourInGroup",
			"from": "#14:2",
			"to": "#15:7"
		}, {
			"@rid": "#16:25",
			"@class": "FlavourInGroup",
			"from": "#14:7",
			"to": "#15:4"
		}]
	}
}
```

**Comment by maggiolo00** [comment]
@lvca 

i'm implementing this on Studio. I have one question. What if i have a property on the edge called from or to?

**Comment by lvca** [comment]
Good question. Do you prefer I rename them as `@from` and `@to` so they are metadata? Makes a lot of sense to me.

**Comment by tglman** [comment]
Hi,
As today is not possible have a property on the edge called from or to that is not the one that have that meaning, not sure is a good idea add a new keyword for this specific case.

**Comment by lvca** [comment]
My idea was to separate attributes by metadata. Since `from` and `to` are not edge's attributes, but they are semantically metadata, it's logical for me using `@`.

**Comment by maggiolo00** [comment]
I suggest to keep in/out instead of renaming them.

**Comment by lvca** [comment]
@luigidellaquila WDYT ?

**Comment by luigidellaquila** [comment]
That's absolutely reasonable for me.
(just let me know, I'll update GoG ;-) )
Thanks

Luigi

**Comment by tglman** [comment]
Hi,

Sorry i may missed my point a bit, and it was that we should not change the in/out to @from/@to here in this context, as well this issue seems target 2.2.x ... that it's an hotfix, and this seems not a fix.

In any case i'm not against to rename in/out to @from/@to, but it shoudl only in the context of 3.0 and probably evaluating similar naming for out_relation and in_relation field of vertex to similar conventions, with the relative changes in all the database internal api.

If we do this change it should be coherent everywhere, would be a nightmare maintain two different names in two different contexts.

**Comment by lvca** [comment]
this command has been released few days before 2.2 and it's completely new and not even documented, so it's not a problem changing it now.

**Comment by maggiolo00** [comment]
Restored in/out

https://github.com/orientechnologies/orientdb/commit/4dd6e62582a229a0d985c496e410395fa1d17cd3
